lib/cocoapods/installer/user_project_integrator.rb

```diff
@@ -112,8 +115,38 @@ def create_workspace
       # @return [void]
       #
       def integrate_user_targets
-        targets_to_integrate.sort_by(&:name).each do |target|
-          TargetIntegrator.new(target).integrate!
+        target_integrators = targets_to_integrate.sort_by(&:name).map do |target|
+          TargetIntegrator.new(target)
+        end
+
+        Config.instance.with_changes(:silent => true) do
+          deintegrator = Deintegrator.new
+          all_project_targets = target_integrators.map(&:user_project).flat_map(&:native_targets).uniq
+          all_native_targets = target_integrators.flat_map(&:native_targets).uniq
+          targets_to_deintegrate = all_project_targets - all_native_targets
```

Comment:
not too worried about that since the link with first target thing is going away in the new DSL